### PR TITLE
Feature: Add task top/end of stack to task info report

### PR DESCRIPTION
**Pull request by shreyasbharath**
This is super useful if you want to generate a stack trace of all the tasks in your system, whether there's a hardfault or a watchdog etc.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by RichardBarry** [comment]
Thanks for this.

Will the reviewer/merger note pxEndOfStack is only available when the following preprocessor condition is met:

#if ( ( portSTACK_GROWTH > 0 ) || ( configRECORD_STACK_HIGH_ADDRESS == 1 ) )

.... so access to pxEndOfStack needs to be likewise guarded in both the updated .c and .h files.  Then please test with both portSTACK_GROWTH set to 1 and -1, and configRECORD_STACK_HIGH_ADDRESS set to both 1 and 0 to catch any accidental compilation errors.

Very few processors have portSTACK_GROWTH set to 1 - I think the PIC24 is does.

**Comment by shreyasbharath** [comment]
Thank you for the response, I'll make the changes this weekend and update the PR.

**Comment by n9wxu** [comment]
> Very few processors have portSTACK_GROWTH set to 1 - I think the PIC24 is does.

There are 3 in our repo.  PIC24, PIC18 and Cygnal

File: include/task.h [hunk 1]
```diff
@@ -144,6 +144,8 @@ typedef struct xTASK_STATUS
 	UBaseType_t uxBasePriority;		/* The priority to which the task will return if the task's current priority has been inherited to avoid unbounded priority inversion when obtaining a mutex.  Only valid if configUSE_MUTEXES is defined as 1 in FreeRTOSConfig.h. */
 	uint32_t ulRunTimeCounter;		/* The total run time allocated to the task so far, as defined by the run time stats clock.  See http://www.freertos.org/rtos-run-time-stats.html.  Only valid when configGENERATE_RUN_TIME_STATS is defined as 1 in FreeRTOSConfig.h. */
 	StackType_t *pxStackBase;		/* Points to the lowest address of the task's stack area. */
+	StackType_t *pxTopOfStack;		/* Points to the top address of the task's stack area. */
```
Comment (n9wxu) [inline include/task.h:147]: Please wrap these two lines with checks for configRECORD_STACK_HIGH_ADDRESS==1 and portSTACK_GROWTH >0.

File: tasks.c [hunk 2]
```diff
@@ -3684,6 +3684,8 @@ static void prvCheckTasksWaitingTermination( void )
 		pxTaskStatus->pcTaskName = ( const char * ) &( pxTCB->pcTaskName [ 0 ] );
 		pxTaskStatus->uxCurrentPriority = pxTCB->uxPriority;
 		pxTaskStatus->pxStackBase = pxTCB->pxStack;
+		pxTaskStatus->pxTopOfStack = (StackType_t *)pxTCB->pxTopOfStack;
```
Comment (n9wxu) [inline tasks.c:3687]: Please wrap these two lines with checks for configRECORD_STACK_HIGH_ADDRESS==1 and portSTACK_GROWTH>0.

The code must compile when pxTopOfStack and pxEndOfStack are not present in the StackType_t structure.

**Comment by n9wxu** [comment]
There are a few conflicts and some conditional compiles need to be added so the code will compile clean when the new pointers are not present.

**Comment by n9wxu** [comment]
Closing this PR and replacing it with #436 .  The effort to bring this PR up to date was too much.  Thank you for the submission and I made sure to include the original author in the new PR.

**Pull request closed without merging**